### PR TITLE
Destination redshift: remove standard inserts

### PR DESCRIPTION
**Pull request by edgao**
closes https://github.com/airbytehq/airbyte-internal-issues/issues/7759

### CODE REVIEW

**Comment by edgao** [comment]
* **#38886** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38886?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @edgao and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by evantahler — COMMENTED** [review]
This will be a breaking change - don't forget that metadata update

File: airbyte-integrations/connectors/destination-redshift/metadata.yaml [hunk 1]
```diff
@@ -24,6 +24,10 @@ data:
           This version introduces [Destinations V2](https://docs.airbyte.com/release_notes/upgrading_to_destinations_v2/#what-is-destinations-v2), which provides better error handling, incremental delivery of data for large syncs, and improved final table structures. To review the breaking changes, and how to upgrade, see [here](https://docs.airbyte.com/release_notes/upgrading_to_destinations_v2/#quick-start-to-upgrading). These changes will likely require updates to downstream dbt / SQL models, which we walk through [here](https://docs.airbyte.com/release_notes/upgrading_to_destinations_v2/#updating-downstream-transformations).
           Selecting `Upgrade` will upgrade **all** connections using this destination at their next sync. You can manually sync existing connections prior to the next scheduled sync to start the upgrade early.
         upgradeDeadline: "2024-03-15"
+      3.0.0:
+        message: >
+          This version removes support for the standard inserts mode. S3 staging was always preferred for being faster and less expensive; as part of Airbyte 1.0, we are officially removing standard inserts.
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-redshift/metadata.yaml:29]:
```suggestion
          Version 3.0.0 of destination-redshift removes support for the "standard inserts" mode. S3 staging was always preferred for being faster and less expensive, and w as part of Airbyte 1.0, we are officially removing the inferior "standard inserts" mode.
```

File: airbyte-integrations/connectors/destination-redshift/metadata.yaml [hunk 2]
```diff
@@ -24,6 +24,10 @@ data:
...
+        upgradeDeadline: "2024-07-31"
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-redshift/metadata.yaml:30]: Before Airbyte 1.0 👍

**Review by evantahler — APPROVED** [review]

File: airbyte-integrations/connectors/destination-redshift/build.gradle [hunk 3]
```diff
@@ -24,7 +24,7 @@ compileKotlin {
 airbyteJavaConnector.addCdkDependencies()
 
 application {
-    mainClass = 'io.airbyte.integrations.destination.redshift.RedshiftDestination'
+    mainClass = 'io.airbyte.integrations.destination.redshift.RedshiftStagingS3Destination'
```
Comment (gisripa) [inline airbyte-integrations/connectors/destination-redshift/build.gradle:27]: Probably rename it to just `RedshiftDestination` ?

File: airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftS3StagingInsertDestinationAcceptanceTest.java [hunk 4]
```diff
@@ -4,21 +4,250 @@
...
 
 /**
  * Integration test testing {@link RedshiftStagingS3Destination}. The default Redshift integration
  * test credentials contain S3 credentials - this automatically causes COPY to be selected.
  */
+// these tests are not yet thread-safe, unlike the DV2 tests.
+@Execution(ExecutionMode.SAME_THREAD)
 @Disabled
-public class RedshiftS3StagingInsertDestinationAcceptanceTest extends RedshiftDestinationAcceptanceTest {
+public class RedshiftS3StagingInsertDestinationAcceptanceTest extends JdbcDestinationAcceptanceTest {
```
Comment (gisripa) [inline airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftS3StagingInsertDestinationAcceptanceTest.java:47]: 😕   why does a disabled test has large diff, rename of some other file ?

File: docs/integrations/destinations/redshift-migrations.md [hunk 5]
```diff
@@ -6,9 +6,21 @@ This version introduces [Destinations V2](/release_notes/upgrading_to_destinatio
...
-- Removal of sub-tables for nested properties
-- Removal of SCD tables
+* Per-record error handling
+* Clearer table structure
+* Removal of sub-tables for nested properties
+* Removal of SCD tables
 
 Learn more about what's new in Destinations V2 [here](/using-airbyte/core-concepts/typing-deduping).
+
+## Upgrading to 3.0.0
```
Comment (gisripa) [inline docs/integrations/destinations/redshift-migrations.md:16]: I think the QA check cares for the order with latest as top and 2.0.0 in the bottom.

**Review by gisripa — APPROVED** [review]
lgtm

File: airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftS3StagingInsertDestinationAcceptanceTest.java (hunk 4 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftS3StagingInsertDestinationAcceptanceTest.java:47]: collapsing the class hierarchy - there used to be base test + standard insert test + s3 test, now there's just the s3 test

File: airbyte-integrations/connectors/destination-redshift/build.gradle (hunk 3 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-redshift/build.gradle:27]: done

**Review by stephane-airbyte — APPROVED** [review]

**Comment by assaadhjb** [comment]
Hello @edgao , Airbyte Cloud user here, May I ask in the roadmap where this has been determined?

Also is it normal that as of today I do not have access to standard INSERT indeed, only COPY is available for me?

Screenshot 2024-07-08 at 14 28 23


Thanks 🙏

**Comment by assaadhjb** [comment]
https://github.com/airbytehq/airbyte/issues/41031

**Comment by edgao** [comment]
sorry, I missed your comment earlier - this was mostly an internal decision, hence us setting a somewhat longer upgrade timeline. Standard inserts usage is unfortunately too low to justify the amount of engineering and support time it takes, and we decided that rather than try to build newer features on both code paths, it made more sense to just stop supporting the inserts mode

**Comment by mycotics** [comment]
This was a terrible idea.

You removed standard inserts without introducing the option to use `iam_role`s for COPY query. 
not only iam_role is the best practice (and the only options in some industries) and if one uses 2FA which requires the access key and secret to be temporary, makes the connector unusable

**Pull request merged**